scipy/spatial/tests/test_spherical_voronoi.py

```diff
@@ -170,3 +229,33 @@ def test_single_hemisphere_handling(self):
             dots = np.einsum('ij,ij->i', sv.vertices, triangles[:, 0])
             circumradii = np.arccos(np.clip(dots, -1, 1))
             assert np.max(circumradii) > np.pi / 2
+
+    @pytest.mark.parametrize("radius", [0.5, 1, 2])
+    @pytest.mark.parametrize("center", [(0, 0, 0), (1, 2, 3)])
+    def test_area_reconstitution(self, radius, center):
+        for points in [self.points, self.hemisphere_points,
+                       self.hemisphere_points2]:
+            sv = SphericalVoronoi(radius * points + center,
+                                  radius=radius,
+                                  center=center)
+            areas = sv.calculate_areas()
+            assert_almost_equal(areas.sum(), 4 * np.pi * radius**2)
+
+    @pytest.mark.parametrize("radius", [0.5, 1, 2])
+    def test_area_reconstitution_large_input(self, radius):
+        np.random.seed(0)
+        n = 1000
```

Comment:
bumping the number of generators to the point where the test fails (v. small Vor regions/ problem angles) might be something I should try locally & compare with discretization approach to see if the limits are similar

Reply:
I like this plan.

Reply:
I have tried random inputs containing up to 1 million points without failure.  I ran out of memory when trying 10 million input points.  In order to vectorize the code, I have to keep the triangulation of the Voronoi regions in memory.  If we want to reduce memory without sacrificing performance I think we will need to use Cython.